deps.bzl

```diff
@@ -1,10 +1,33 @@
-load("//private:version_repo.bzl", "version_repo")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
 load("//private:globals.bzl", "GLOBALS")
 load("//private:globals_repo.bzl", "globals_repo")
+load("//private:version_repo.bzl", "version_repo")
 
-def bazel_features_deps():
+def bazel_features_deps(legacy = False):
     version_repo(name = "bazel_features_version")
     globals_repo(
         name = "bazel_features_globals",
         globals = GLOBALS,
     )
+
+    if legacy:
+        maybe(
+            http_archive,
+            name = "bazel_skylib",
```

Comment:
Since we are loading this in the top-level package where we also provide the `.bzl` files, this is now a hard requirement that should be loaded unconditionally.

Reply:
If `legacy` is `False`, then bzlmod is enabled. `bazel_skylib` is loaded in the `MODULE.bazel`. If `legacy` is `True`, we use the `maybe` so that the client can specify their preferred version of `bazel_skylib`. In either scenario, `bazel_skylib` will be loaded.

Background: The `legacy` boolean exists because `bazel_features_deps` is called from the bzlmod extension and from a `WORKSPACE`.

Reply:
If `legacy` is `True` (Bzlmod isn't used), where will bazel_skylib be loaded if the downstream project doesn't have a direct dependency on it? That's going to be very rare, but we should still support it.

Reply:
> If legacy is True (Bzlmod isn't used),

Yes. Sorry. Dumb typo.

I see. You are suggesting that the default behavior is for `legacy = True`. That  makes sense.